api/class2.js

```diff
@@ -22,66 +25,211 @@ router.get('/customers', function(req, res) {
+//     if (err) {
+//       console.log('ERROR fetching from the database with surname:', err);
+//       return;
+//     }
+//     console.log('Request succeeded, new data fetched', rows);
+//     res.status(200).json({
+//       customers: rows
+//     });
+//   });
+// });
 
-  // TODO: add code here
-});
 
 
-router.put('/customers/:id', function(req, res) {
-  // EXPECTED JSON Object:
-  // {
-  //   title: 'Mr',
-  //   firstname: 'Laurie',
-  //   surname: 'Ainley',
-  //   email: 'laurie@ainley.com'
-  // }
+router.post('/customers/', function (req, res) {
+  // TODO: add code here
+  var sql = `insert into customers (title, first_name, surname, email)values('${req.body.title}', '${req.body.firstname}', '${req.body.surname}', '${req.body.email}')`;
+
+  db.run(sql, [], (err, rows) => {
+    if (err) {
+      console.log('ERROR fetching from the database:', err);
+      return;
+    }
+    console.log('Request succeeded, new data fetched', rows);
+    res.status(200).json({
+      customers: rows
+    });
+  });
+});
 
+router.put('/customers/:id', function (req, res) {
   // TODO: add code here
+  //if all the info wants to be updated
+  var id = req.params.id
+  const {
+    title,
+    first_name,
+    surname,
+    email
+  } = req.body
+  var sql = `update customers set 
+  title='${title}',
+  first_name='${first_name}',
+  surname='${surname}',
+  email='${email}' 
+  where id='${id}'`;
+
+
+  //if just part of it wants to be updated
+  var sql1 = `update customers set 
+  first_name='${first_name}'
+  where id='${id}'`;
+  db.run(sql1, [], (err, rows) => {
+    if (err) {
+      console.log('ERROR fetching from the database:', err);
+      return;
+    }
+    console.log('Request succeeded, new data fetched', rows);
+    res.status(200).json({
+      customers: rows
+    });
+  });
 });
 
 
+
+
 // get '/reservations'
 // TODO: add code here
 
+router.get('/reservations', function (req, res) {
+  // TODO: add code here
+  var sql = `select * from reservations`;
 
+  db.all(sql, [], (err, rows) => {
+    if (err) {
+      console.log('ERROR fetching from the database with surname:', err);
+      return;
+    }
+    console.log('Request succeeded, new data fetched', rows);
+    res.status(200).json({
+      reservations: rows
+    });
+  });
+});
 // get '/reservations/:id'
 // TODO: add code here
+router.get('/reservations/:id', function (req, res) {
+  var id = req.params.id;
+  var sql = `select * from reservations where id=${id}`;
+  console.log(sql);
+  db.all(sql, [], (err, rows) => {
+    if (err) {
+      console.log('ERROR fetching from the database:', err);
+      return;
+    }
+    console.log('Request succeeded, new data fetched', rows);
+    res.status(200).json({
+      reservations: rows
+    });
+  });
+});
+
+//delete '/reservations/:id'
+
+router.delete('/reservations/:id', function (req, res) {
+  var id = req.params.id;
+  var sql = `delete from reservations where id=${id}`;
+  db.all(sql, [], (err, rows) => {
```

Comment:
When you're not retrieving data from the database, you should be using db.run not db.all.